.gitlab-ci.yml

```diff
@@ -1,86 +1,175 @@
+# Gitlab-CI Workflow
+# stages:
+#   build:
+#     - Runs on commits on master or tags that match the pattern "v[0-9]+\.[0-9]+.*$". (e.g. 1.0, v2.1rc1)
+#   deploy-staging: 
+#     - Runs on commits on master or tags that match the pattern v1.0, v2.1rc1 (continues deployment)
+#   deploy-production:
+#     - Runs on tags that match the pattern v1.0, v2.1rc1 (manual deployment)
+
 variables:
-  BACKEND_CONTAINER_REPO:         "docker.io/parity/substrate-telemetry-backend"
-  FRONTEND_CONTAINER_REPO:        "docker.io/parity/substrate-telemetry-frontend"
-  BACKEND_IMAGE_FULL_NAME:        "${BACKEND_CONTAINER_REPO}:${CI_COMMIT_SHORT_SHA}-beta"
-  FRONTEND_IMAGE_FULL_NAME:       "${FRONTEND_CONTAINER_REPO}:${CI_COMMIT_SHORT_SHA}-beta"
-  KUBE_NAMESPACE:                 "substrate-telemetry"
-  VAULT_SERVER_URL:               "https://vault.parity-mgmt-vault.parity.io"
-  VAULT_AUTH_PATH:                "gitlab-parity-io-jwt"
-  VAULT_AUTH_ROLE:                "cicd_gitlab_parity_${CI_PROJECT_NAME}"
+  # Build Variables (Mandatory)
+  CONTAINER_REPO:             ""
+  DOCKERFILE_DIRECTORY:       ""
+  
+  # Deploy Variables (Mandatory)
+  HELM_NAMESPACE:             "substrate-telemetry"
+  HELM_RELEASE_NAME:          "substrate-telemetry"
+  HELM_CHART:                 "parity/substrate-telemetry"
+  
+  # Deploy Variables (Optional)
+  HELM_REPO_NAME:             "parity"
+  HELM_REPO_URL:              "https://paritytech.github.io/helm-charts/"
+  HELM_CONFIGMAP_NAME:        "helm-custom-values"
+  HELM_CONFIGMAP_KEYNAME:     "values-parity.yaml"
+
+  # Manual Variables (Optional)
+  ## Could be used in the webconsole when triggering the pipeline manually
+  ## DO NOT SET THEM IN THIS FILE!! They've been mentioned here only for documentation purposes!
+  FORCE_DEPLOY:               ""        # boolean: true or false - triggers the deploy-production stage
+  FORCE_DOCKER_TAG:           ""        # choose an existing docker tag to be deployed (e.g. v1.2.3)
+
+  # Vault variables (Optional)
+  VAULT_SERVER_URL:           "https://vault.parity-mgmt-vault.parity.io"
+  VAULT_AUTH_PATH:            "gitlab-parity-io-jwt"
+  VAULT_AUTH_ROLE:            "cicd_gitlab_parity_${CI_PROJECT_NAME}"
+
+
+default:
+  before_script:
+    - |-
+      echo defining DOKCER_IMAGE_TAG variable
+      if [[ $FORCE_DOCKER_TAG ]]; then
+        export DOCKER_IMAGE_TAG="${FORCE_DOCKER_TAG}"
+      elif [[ $CI_COMMIT_TAG =~ ^v[0-9]+\.[0-9]+.*$ ]]; then
+        export DOCKER_IMAGE_TAG="${CI_COMMIT_TAG}"
+        #export BUILD_LATEST_IMAGE="true"
```

Comment:
Is this supposed to be commented out? There's a reference to it below, but I can't see how the variable would otherwise exist (unless it comes from the UI, and if so maybe it's worth documenting as you have done FORCE_DEPLOY etc?) :)

Reply:
I've commented it out intentionally, until the production is not ready we shouldn't change overwrite the latest tag in the docker repository.
once everything is tested and production env is prepared we can uncomment this and publish the new docker image as the latest tag.